extensions/cornerstone/src/panels/PanelMeasurement.tsx

```diff
@@ -32,69 +42,50 @@ export default function PanelMeasurementTable({
-    measurementService.remove(uid);
+    displayMeasurements.forEach(m => (m.isActive = m.uid === uid));
   };
 
-  const renameMeasurement = (uid: string) => {
-    jumpToImage(uid);
-    const labelConfig = customizationService.get('measurementLabels');
-    const measurement = measurementService.getMeasurement(uid);
-    showLabelAnnotationPopup(measurement, uiDialogService, labelConfig).then(val => {
-      measurementService.update(
-        uid,
-        {
-          ...val,
-        },
-        true
-      );
-    });
-  };
-
-  const changeColorMeasurement = (uid: string) => {
-    const { color } = measurementService.getMeasurement(uid);
-    const rgbaColor = {
-      r: color[0],
-      g: color[1],
-      b: color[2],
-      a: color[3] / 255.0,
-    };
-    colorPickerDialog(uiDialogService, rgbaColor, (newRgbaColor, actionId) => {
-      if (actionId === 'cancel') {
-        return;
+  const bindCommand = (name: string, callJump = false) => {
+    return (uid: string) => {
+      if (!uid) {
+        debugger;
       }
-
-      const color = [newRgbaColor.r, newRgbaColor.g, newRgbaColor.b, newRgbaColor.a * 255.0];
-      // segmentationService.setSegmentColor(viewportId, segmentationId, segmentIndex, color);
-    });
+      if (callJump) {
+        commandsManager.runCommand('jumpToMeasurement');
+      }
+      commandsManager.runCommand(name, { uid });
+      if (callJump === null) {
+        onMeasurementItemClickHandler(uid, true);
+      }
+    };
   };
 
-  const toggleLockMeasurement = (uid: string) => {
-    measurementService.toggleLockMeasurement(uid);
-  };
+  const jumpToImage = bindCommand('jumpToMeasurement', null);
+  const removeMeasurement = bindCommand('removeMeasurement');
+  const renameMeasurement = bindCommand('renameMeasurement', true);
+  const toggleLockMeasurement = bindCommand('toggleLockMeasurement');
+  const toggleVisibilityMeasurement = bindCommand('toggleVisibilityMeasurement');
 
-  const toggleVisibilityMeasurement = (uid: string) => {
-    measurementService.toggleVisibilityMeasurement(uid);
-  };
+  const additionalFilter = filterAdditionalFinding(measurementService);
 
-  const measurements = displayMeasurements.filter(
-    dm => dm.measurementType !== measurementService.VALUE_TYPES.POINT && dm.referencedImageId
-  );
-  const additionalFindings = displayMeasurements.filter(
-    dm => dm.measurementType === measurementService.VALUE_TYPES.POINT && dm.referencedImageId
-  );
+  useEffect(() => {
+    const { measurementFilter: trackedFilter, untrackedFilter } = measurementFilters;
+    setMeasurements(
+      displayMeasurements.filter(item => !additionalFilter(item) && trackedFilter(item))
+    );
+    setAdditionalFindings(
+      displayMeasurements.filter(item => additionalFilter(item) && trackedFilter(item))
+    );
+    setUntrackedFindings(displayMeasurements.filter(untrackedFilter.bind(measurementFilters)));
+  }, [measurementFilters, displayMeasurements]);
```

Comment:
You don't need a state for these; they can all be computed during the render call. If you remove `useEffect` and those `useState` hooks, it should work. The `useMeasurements` hook already abstracts the reactivity of the measurement service, so everything else can be computed states (not actual states).